presto-hive/src/main/java/io/prestosql/plugin/hive/rubix/RubixInitializer.java

```diff
@@ -125,7 +125,10 @@ public void stopRubix()
     {
         try (Closer closer = Closer.create()) {
             closer.register(() -> {
-                if (bookKeeperServer != null && bookKeeperServer.isServerUp()) {
+                if (bookKeeperServer != null) {
+                    // This might throw NPE if Thrift server hasn't started yet (it's initialized
+                    // asynchronously from BookKeeperServer thread).
+                    // TODO: improve stopping of BookKeeperServer server in Rubix
```

Comment:
Maybe catch the NPE for now?

Reply:
> Maybe catch the NPE for now?

In production this shouldn't happen as `bookKeeperServer` will initialize thrift server almost immediately. Shutdown of `Connector` also doesn't terminate Presto.

In tests this NPE is used to retry `stopRubix`, otherwise ports would leak.